### DISCUSSION TITLE
Localize pages folders with rewrites.

### DISCUSSION THREAD

**Discussion by jurajkrivda** [vercel/next.js#18666] [Help]
Hi. I have a Spanish version of the site, and I want to keep the files in the /pages folder in English because of my conventions. But in the url I want to have a Spanish slug, eg the folder is called /articles and I have index.tsx in it, but in the url I want /articulos. I was thinking about using rewrites eg:

```
async rewrites () {
     return [
       {
         source: '/articulos',
         destination: '/articles',
       },
       {
         source: '/articulos/: id',
         destination: '/articles/: id',
       },
     ]
   },
```

Do you think it's okay? Do you also have English names in the pages folder? Is it possible to merge this article rule into one object? Thank you.

**Comment by mimoid-prog** [comment]
There is a discussion about this [https://github.com/vercel/next.js/discussions/18572](https://github.com/vercel/next.js/discussions/18572)
